### ISSUE TITLE
Rethink z-indexes

### ISSUE THREAD

**Issue by dimaip** [neos/neos-ui#1382]
We've merged a change to refactor usage of z-indexes: https://github.com/neos/neos-ui/pull/1361
However it caused a few regressions, e.g. see the screenshot.
The idea was to group z-indexes by their stacking contexts and stop making up their values arbitrarily.
However the problem is that currently we group z-indexes not always by the stacking contexts, which causes various issues.
So we need to fix those regressions in a clever way. In most cases it should be enough to force new stacking contexts where needed.

![image](https://user-images.githubusercontent.com/837032/33530482-240afdf8-d891-11e7-9f2a-e85206b90a77.png)

**Comment by dimaip** [comment]
Maybe @Duske you have an opinion?

**Comment by Duske** [comment]
I think introducing new stacking contexts would be most straightforward approach.
Currently I have no time to look into it in more detail 🙁

**Comment by dimaip** [comment]
Yep, apparently there's even a dedicated property to do just that: https://developer.mozilla.org/en-US/docs/Web/CSS/isolation
We need to rethink a system that would handle nested z-indexes, essentially it should be a nested object.

E.g. this
```js
{
 primaryToolbar: {
   dropDown: true
 },
 contentCanvas: {
   some: {
     nesting: true,
     sibling: true
   }
 }
}
```
would transform into:

```js
--zIndex-primaryToolbar: 1
--zIndex-primaryToolbar-dropDown: 1
--zIndex-contentCanvas: 2
--zIndex-contentCanvas-some: 1
--zIndex-contentCanvas-some-nesting: 1
--zIndex-contentCanvas-some-sibling: 2
```
And of course this structure should resemble actual stacking contexts and don't be arbitrary.

**Comment by jonnitto** [comment]
Also a topic if you have a tooltip with a following select box  
<img width="298" alt="screenshot 2017-12-22 16 11 40" src="https://user-images.githubusercontent.com/4510166/34302837-f9475fb6-e732-11e7-984d-c65b74a5f947.png">

**Comment by johannessteu** [comment]
Those should be rendered inline/relative. At least i had it in my tooltip change. NOt sure what way @dimaip re-implemented those as the PR was closed. But in general those should fit in between and don't overlay the next property wich might has an validation error as well

**Comment by dimaip** [comment]
See https://github.com/neos/neos-ui/issues/1426

**Comment by mstruebing** [comment]
Should be fixed by #1937

**Comment by dimaip** [comment]
This particular bug is fixed, but the idea of managing z-indexes still needs more love.
